### PR TITLE
Feature: Replace `clean` commands with `reset` and `purge` commands

### LINKED ISSUE #71: Add more granular clean commands
> I’m working on a rollup and want to restart the sequencer from a clean state but don’t want to lose my env changes. astria-go dev clean alone does not work because you lose the cometbft configs. I have to run astria-go dev init to get them back

We could add more granular clean commands to solve this issue.
```
clean config
clean state
clean env
clean binaries
clean all
```

This would let the developer just clear state without losing any manual env changes or cometbft genesis changes

### PR DESCRIPTION
**Pull request by sambukowski**
The existing workflow for initializing, cleaning/reseting data, and running or re-running for the cli wasn't particularly clear. Specifically the user flow for when you needed to re-run `init` after a given command and that running `init` would reset all config.
`init` now ignores all files that currently exist, so a user can make changes to their env files or other config and re-run `init` without fear of those changes being removed.
The existing `clean` commands have been removed and replace with more descriptive `reset` and `purge` commands.
The new user flow is as follows: 
- if you run a `reset` command, you don't need to run `init` again before `run`.
- if you run a `purge` command, you do need to run `init` again before `run`

`reset` commands return the data to their initial state, as though `init` was run for the first time.
`purge` commands simply delete the specified data

Added commands:
`dev reset config` -> reset cometbft configs
`dev reset env` -> reset env files
`dev reset state` -> reset the state
`dev purge binaries` -> delete the downloaded binaries
`dev purge all` -> delete an instance

closes #71

### CODE REVIEW

File: cmd/devtools/purge.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/astria/astria-cli-go/cmd"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// resetCmd represents the root reset command
```
Comment (steezeburger) [inline cmd/devtools/purge.go:13]: update comment

File: cmd/devtools/purge.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	// subcommands
+	purgeCmd.AddCommand(purgeBinariesCmd)
+	purgeCmd.AddCommand(purgeAllCmd)
+}
+
+// purgeBinariesCmd represents the 'purge binaries' command
+var purgeBinariesCmd = &cobra.Command{
+	Use:    "binaries",
+	Short:  "Delete all binaries for a given instance.",
+	Long:   "Delete all binaries for a given instance.",
```
Comment (steezeburger) [inline cmd/devtools/purge.go:35]: I don't think you need to include both Short and Long if they're the same

File: cmd/devtools/purge.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	Use:    "purge",
+	Short:  "The root command for purging the local development instance data.",
+	Long:   `The root command for purging the local development instance data. Whenever a purge command is run, it will delete the specified data. You will need to re-initialize the instance to replace the data.`,
+	PreRun: cmd.SetLogLevel,
+}
+
+func init() {
+	// top level command
+	devCmd.AddCommand(purgeCmd)
+	purgeCmd.PersistentFlags().StringP("instance", "i", DefaultInstanceName, "Choose the target instance for resetting.")
```
Comment (steezeburger) [inline cmd/devtools/purge.go:24]:
```suggestion
	purgeCmd.PersistentFlags().StringP("instance", "i", DefaultInstanceName, "Choose the target instance for purging.")
```

File: cmd/devtools/reset.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+		fmt.Println("Error removing file:", err)
+		return
+	}
+	err = os.Remove(filepath.Join(localConfigDir, DefaultCometbftValidatorFilename))
+	if err != nil {
+		fmt.Println("Error removing file:", err)
+		return
+	}
+
+	recreateCometbftAndSequencerGenesisData(localConfigDir)
```
Comment (steezeburger) [inline cmd/devtools/reset.go:72]: We should reorganize where this function lives now that it's shared.

Maybe we move all the embedded and config file functionality to its own file in `cmd/devtools`, maybe `confighelpers.go` or something?

Comment (steezeburger) [reply cmd/devtools/reset.go:72]: Tangential, but I noticed a few misspellings, `instancDir`, that you can fix when you move things, but I can't comment on that section of the code.

**Review by steezeburger — CHANGES_REQUESTED** [review]

File: cmd/devtools/init.go [hunk 5]
```diff
@@ -402,3 +422,30 @@ func replaceInFile(filename, oldValue, newValue string) error {
 
 	return nil
 }
+
+// copyFile copies the contents of the src file to dst.
+// If dst does not exist, it will be created, and if it does, it will be overwritten.
+func copyFile(src, dst string) error {
```
[the remark this answers is not in the document]
Comment (sambukowski) [reply cmd/devtools/init.go:428]: Both `src`, and `dst` are strings. After doing a little digging it seems this is the more idiomatic way of having multiple arguments of the same type in go. But we're also not being consistent with this in many places and should probably update that.

[the remark this answers is not in the document]
Comment (steezeburger) [reply cmd/devtools/init.go:428]: Why would src be anything but a string? That's what os.Open takes.

**Review by steezeburger — CHANGES_REQUESTED** [review]

**Review by steezeburger — APPROVED** [review]

**Pull request merged**